### ISSUE TITLE
bokutachinogothic2 is mis-centered and cut off (japanese)

### ISSUE THREAD

**Issue by Ulti-FD** [TeamRizu/Texture-Font-Generator-2020-Squirrel#2]
https://cdn.discordapp.com/attachments/357630502296879104/767587688403828756/Boku2-Regular.otf
https://cdn.discordapp.com/attachments/357630502296879104/767587624515797052/Boku2-Bold.otf
On both this new (jap) generator and older versions of the (jap) generator, this font is malfunctioning. The output is not properly aligned, which also causes pixels in the bottom to cut off.
Take note of the g and j.
![font](https://user-images.githubusercontent.com/24206396/96400213-ae8f8880-11bf-11eb-9ad4-0985cf275632.png)
These two texts are not supposed to be clipping into eachother, but they are since the spritesheet is misaligned.
![](https://cdn.discordapp.com/attachments/760945361157488682/767561421050871868/unknown.png)
The problem seems to be with this font itself as I cannot reproduce this issue with another font, including other japanese fonts. I'm making an issue here because other programs can render this font perfectly fine. The cutoff is also present in ms paint and the truetype font preview, but everything else I've found handles it correctly, so I believe it's an issue with both the program and the font.
Messing with the alignment settings (blue and yellow bars), padding, anti-aliasing, and size does not seem to affect this.

**Comment by Scraticus** [comment]
Hiya! thanks for using the new font generator.

This actually isn't a fault with the font generator in the case of those fonts. The original creator actually has clipped the sub part of the font, thus causing this weirdness.

![image](https://user-images.githubusercontent.com/11047768/96465040-6c4a6380-1220-11eb-9b6e-946041433a9e.png)

This is how windows sees the font, and you can see it has a non-standard width on a few of the glyphs, as well as a short 'bottom'

![image](https://user-images.githubusercontent.com/11047768/96464715-0fe74400-1220-11eb-9fdc-b8fe471786f0.png)

We can only generate what was added by the official glyph designer, so nothing we can do with this one sadly.

**Comment by Ulti-FD** [comment]
Strange. At some point, this font was used to make the text used in the original spawncamping-wallhack theme. Those spritesheets have no cutoff and correct alignment. Either some super ancient version of the jap text generator was used that somehow deals with the font better or Prim has some version of the font I can't find.

**Comment by Scraticus** [comment]
I'm having a look through the history of SCWH, to see if we can match the font glyphs for you, I believe the earlier versions of this fontset were brought in from a different font which was made by google, and this doesn't have the clipping.

The font you need is Source Han Sans Regular, and Source Han Sans Bold, which are a (non-clipped) close version. 

![image](https://user-images.githubusercontent.com/11047768/96511467-07f6c680-1257-11eb-8976-a98af7f91708.png)

Your original screenshot vs the new font:

new
![image](https://user-images.githubusercontent.com/11047768/96512253-21e4d900-1258-11eb-9fe5-7a7acfd093d0.png)

![image](https://user-images.githubusercontent.com/11047768/96513065-6ae95d00-1259-11eb-9fdb-96ca41bbadfb.png)


You will need to tweak the padding to suit your uses, and hopefully this helps you with your theme work!

I've [linked](https://drive.google.com/file/d/1VY8C0r3yOJE22nVuRDwp3eq2C7Hj9AFQ/view?usp=sharing) the two fonts for you already so you don't need to download a 2GB file. 

They come from [here](https://github.com/adobe-fonts/source-han-sans):

**Comment by Ulti-FD** [comment]
Really appreciate your time, I will be moving forward with this font. I found a medium version of this font thats between the regular and bold that has pretty much the exact same width as bokuG2.
Seems that all my detective work was for nothing though, I messaged Prim about this and he told me he had the same issue when he originally used the font. But he corrected **everything** manually LMAO. I should have asked him before I even looked into all this.
Thank you for the solution regardless. Closing.
